docs/specification.md

```diff
@@ -231,15 +328,49 @@ Multiple annotations can be attached to an element. All annotations have named a
 
 The annotation examples above shows a validation scenario but annotations are not limited to this purpose. The developer has the freedom to extend the code generation tool to leverage annotations for their application's needs. In the `Customer` example, the developer could use these annotations to generate `validate` methods on each of the generated object types.
 
-TODO: Directives
+Directives are used to ensure that an annotation's arguments match an expected format. For constraining a numeric or length range, the directive for `@range` might look like this:
 
-### Default values
+```apexlang
+directive @range(min: u32, max: u32)
+  on FIELD
+  require @valid on TYPE
+```
 
-Fields can also specify default values when a type is initialized. This needs to be carefully considered in the code generation. Languages vary of how this would be implemented.
+This directive specifies parameters but also where it can be applied. The syntax `on FIELD require @valid on TYPE` means "only allow this on fields where there is also an @valid annotation on its type". Multiple locations are supported and delimited with `|`:
```

Comment:
```suggestion
This directive specifies parameters but also where it can be applied. The syntax `on FIELD require @valid on TYPE` means "only allow this on fields where there is also an @validate annotation on its type". Multiple locations are supported and delimited with `|`:
```